src/tyr/route_serializer_osrm.cc

```diff
@@ -12,6 +12,8 @@
 #include "midgard/util.h"
 #include "odin/enhancedtrippath.h"
 #include "odin/util.h"
+#include "odin/narrative_builder_factory.h"
+#include "odin/narrativebuilder.h"
```

Comment:
Having these here is ofc debatable. We have following problem here: Currently we only build instructions during building the maneuver. But we decide the distance when a maneuver is played in the osrm_serializer and the instructions do depend on the distance (e.g. turn right in 200 meters). So the two options in my mind are:
- Calculate the distances here and add the 'IN 200 meters' information here. This needs the dependency and that's how I did it.
- Inject the distance information during maneuver building. So we later on just call here e.g. `verbal_transition_alert_instruction()` and we also get the distance when to play it.

Reply:
yeah i think the long term way we should do it is the second way primarily because it is odins main function, deciding where maneuvers are. deciding when to warn about them is very closely related in my opinion.

having said that, you already did all the work and there has been slow progress on this effort in the past so i think it could stay this way until someone has time for a larger refactor. i would wager that others would agree especially because there are other odin deps mixed in here as well (which was already not great and was for the same reason).

as an aside.. maybe we should have had the actor stuff that tyr provides be part of the valhalla namespace exposed in the main valhalla header and then just have odin subsume all the serialization stuff (essentially dividing up tyrs parts and then removing tyr completely). i think it would be reasonably appropriate for odin to generate maneuvers narrative and do serialization. its essentially serializing the path into maneuvers and then serializing maneuvers into narrative. may as well do the final serialization into json or pbf bytes. this of course ignores that there are some endpoints that have no directions or narrative and maybe those would seem strange calling into odin.. meh just some thoughts :smile:

Reply:
Thanks for having a look! I completely agree. The second way feels more correct, but would need some more time. Would also be better so the normal valhalla json answer could be enhanced by the same distance information which might be helpful.

But then let me clean up this one and write some test. I'll also open a cleanup issue for doing it the second way. Maybe I'll have some time in the future to clean it up.